src/Subscriptions/Storage/RedisStorageManager.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+namespace Nuwave\Lighthouse\Subscriptions\Storage;
+
+use Illuminate\Contracts\Config\Repository;
+use Illuminate\Contracts\Redis\Factory;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Collection;
+use Nuwave\Lighthouse\Subscriptions\Contracts\StoresSubscriptions;
+use Nuwave\Lighthouse\Subscriptions\Subscriber;
+
+/**
+ * Stores subscribers and topics in redis.
+ * - Topics are subscriptions like "userCreated" or "userDeleted".
+ * - Subscribers are clients that are listening to channels like "private-lighthouse-a7ef3d".
+ *
+ * This manager stores a SET of subscriber channels and the subscribers itself like this:
+ * - graphql.topic.userCreated = [ "presence-lighthouse-1", "presence-lighthouse-2", ... ]
+ * - graphql.topic.userDeleted = [ "presence-lighthouse-5", "presence-lighthouse-6", ... ]
+ * - graphql.subscriber.presence-lighthouse-1 = { query: "{ id, name }" }
+ * - graphql.subscriber.presence-lighthouse-2 = { query: "{ name, created_at }" }
+ */
+class RedisStorageManager implements StoresSubscriptions
+{
+    /**
+     * @var string
+     */
+    public const TOPIC_KEY = 'graphql.topic';
+
+    /**
+     * @var string
+     */
+    public const SUBSCRIBER_KEY = 'graphql.subscriber';
+
+    /**
+     * @var \Illuminate\Redis\Connections\Connection
+     */
+    private $connection;
+
+    /**
+     * The time to live for items in the cache.
+     *
+     * @var int|null
+     */
+    protected $ttl;
+
+    public function __construct(Repository $config, Factory $redis)
+    {
+        $this->connection = $redis->connection(
+            $config->get('lighthouse.broadcasters.echo.connection', 'default')
+        );
+        $this->ttl = config('lighthouse.subscriptions.storage_ttl', null);
+    }
+
+    /**
+     * @param  array<string, mixed>  $input
+     * @param  array<mixed>  $headers
+     * @deprecated will be removed in favor of subscriberByChannel
+     */
+    public function subscriberByRequest(array $input, array $headers): ?Subscriber
```

Comment:
Since we are working towards v5 now, we can go ahead and remove this method from here and from the interface.